core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java

```diff
@@ -642,32 +642,6 @@ public void add(Reader reader, String baseURI, RDFFormat dataFormat, Resource...
 		}
 	}
 
-	@Override
-	public void add(Statement st, Resource... contexts) throws RepositoryException {
-		boolean localTransaction = startLocalTransaction();
-		addWithoutCommit(st, contexts);
-		try {
-			conditionalCommit(localTransaction);
-		} catch (RepositoryException e) {
-			conditionalRollback(localTransaction);
-			throw e;
-		}
-	}
-
-	@Override
-	public void add(Iterable<? extends Statement> statements, Resource... contexts) throws RepositoryException {
-		boolean localTransaction = startLocalTransaction();
-		for (Statement st : statements) {
-			addWithoutCommit(st, contexts);
-		}
-		try {
-			conditionalCommit(localTransaction);
-		} catch (RepositoryException e) {
-			conditionalRollback(localTransaction);
-			throw e;
-		}
-	}
-
```

Comment:
I don't think you should remove this. For one thing, `SPARQLConnection` does not actually implement `setNamespace` (because there is no SPARQL command for this), so we can't support the transfer of namespace in this way when adding a Model. 

If this is something we want to support for `SPARQLConnection` as well we should probably look at ensuring that namespaces in the supplied model make it into the serialized SPARQL update that will be sent to the endpoint, somehow. This is handled in the `createInsertDataCommand` method. But to be fair even then I'm not sure you can guarantee it. There is no general standard on whether a remote SPARQL endpoint stores/reuses the the namespace definitions you supply when you do an update.

TL;DR I'd revert this change for now, and just not worry about SPARQLConnection for the time being.